Add dracut config as discovery subpackage

lzap opened the pull request:
We are unable to regenerate dracut image via livecd-creator due to ordering issues. A BZ was raised that keyboard driver is missing for MS Hyper-V VMs. The missing drivers can be only configured before kernel installs, unless I am wrong doing this in %pre should do the trick.

More info at https://issues.redhat.com/browse/SPMM-9540

This is nightly only change, upstream we do not build FDI this way (we use lorax nowdays). Therefore this new subpackage does not need to be whitelisted at all upstream, we need to pull it downstream to fix the problem in brew.

lzap commented:
Build failed, let me rebase and see why.

evgeni commented on packages/plugins/foreman-discovery-image-service/foreman-discovery-image-service.spec:
```diff
@@ -40,7 +47,20 @@ Metapackage with dependencies for FDI text-user interface
+# MS Hyper-V. Therefore the configuration must be dropped before the kernel package
+# executes its %post scriptlet. Doing it in %pre ensures the configuration is created
+# beforehand.
+echo 'add_drivers="mptbase mptscsih mptspi hv_storvsc hid_hyperv hv_netvsc hv_vmbus"' > %{_sysconfdir}/dracut.conf.d/99-discovery.conf
```
Why is this a post script vs just a file shipped in the package?

lzap replied:
So correct me if I am wrong, but the goal is to have that configuration in-place before kernel's %post scriptlet is executed. Since %post scriptlets are executed one by one as packages are installed I cannot guarantee the order (because kernel can/will likely install before this package).

This actually makes me think if I should actually change this to `%pretrans` that is guaranteed to be executed before "transaction", whatever that actually is: https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/

evgeni replied:
Wait, what, the result is dependant on the order dnf decides to install the packages in? What the actual fuck?!

No, I am not going to think about this at this time.

If you're going to edit the file in any scriptlet, please add a `%ghost` entry to the files section, so rpm at least knows who owns the file.

lzap replied:
I have found a solution, %triggerprein is exactly what I need. I am gonna try to test that.

BTW do I miss buildroot on this line? I guess this is correct:

```
echo 'xxx' > %{buildroot}%{_sysconfdir}/dracut.conf.d/99-discovery.conf
```

lzap replied:
Damn this must NOT have a buildroot macro of course! :) Jeez and its Monday.

https://pbs.twimg.com/media/FN9eS68X0AIVaqv.jpg

evgeni reviewed (changes requested):
The package needs to be added to comps if you want to ship it in the repos (which I guess you do).

No whitelisting required.

lzap commented:
Changing to %triggerprein, added %ghost, fixed the missing buildroot macro, whitelisted the subpackage.

Do not merge yet, I am going to test the scratchbuild manually by creating a repository.

lzap commented:
Created repo at https://people.redhat.com/~lzapleta/temp/fdi-dracut/
